subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildIntegrationTest.groovy

```diff
@@ -700,7 +700,7 @@ task inputsAndOutputs {
     outputs.file 'src.a.txt'
     outputs.upToDateWhen { project.hasProperty('uptodate') }
     doFirst {
-        outputs.files.singleFile.text = "[${inputs.files.singleFile.text}]"
+        file('src.a.txt').text = "[${file('src.txt').text}]"
```

Comment:
Nit: would be nice to use local variables to refer to the same files twice.

Reply:
Done.